src/include/platform/OTARequestorInterface.h

```diff
@@ -173,9 +202,11 @@ class OTARequestorInterface
     // Get image update progress in percents unit
     virtual CHIP_ERROR GetUpdateProgress(EndpointId endpointId, chip::app::DataModel::Nullable<uint8_t> & progress) = 0;
 
-    // Get requestor state
+    // Get the value of the UpdateState attribute of the OTA Software Update Requestor Cluster on the given endpoint
     virtual CHIP_ERROR GetState(EndpointId endpointId,
                                 chip::app::Clusters::OtaSoftwareUpdateRequestor::OTAUpdateStateEnum & state) = 0;
+    // Getter for the value of the UpdateState cached by the object
+    virtual UpdateState GetCurrentUpdateState() = 0;
```

Comment:
What is the need for this function? Isn't it pretty much just the same as `GetState` but with some kind of mapped values from `OTAUpdateStateEnum` to `UpdateState`?

Reply:
GetState takes endpointId, the driver has no awareness of endpoints.

Reply:
Taking a closer look, it seems maybe `GetState` is not needed. The caller should just be able to call `OtaRequestorServerGetUpdateState` directly without needing to go through the OTARequestor class. I'll file an issue to look into that.

Reply:
On a separate note, you shouldn't need to create your own `UpdateState` here. Boris is moving the enums out to it's own file which should be used here: https://github.com/project-chip/connectedhomeip/pull/15686. I can also file an issue to track this change.

Reply:
OK